openvdb/openvdb/viewer/Viewer.cc

```diff
@@ -328,7 +328,9 @@ ThreadManager::doView()
     // This function runs in its own thread.
     // The mClose and mRedisplay flags are set from the main thread.
     while (!mClose) {
-        if (mRedisplay.compare_and_swap(/*set to*/false, /*if*/true)) {
+        bool expected = true;
+        mRedisplay.compare_exchange_strong(expected, false);
+        if (expected) {
```

Comment:
Looks correct, but I 'm curious why not
if (mRedisplay.compare_exchange_strong(expected, false))
as it will return true on a successful store; which will correspond to the old behaviour?

Reply:
tbb says  "x.compare_and_swap(y,z) | if x equals z, then do x=y. In either case, return old value of x." https://www.threadingbuildingblocks.org/docs/help/tbb_userguide/Atomic_Operations.html
So in the old version...if `mRedisplay` ....
* was `false` it won't be swapped, so will return `false`.
* was `true` it will be swapped so it will return `true`.
New version:
If it...
* was `false` it won't be swapped (because `expected` is `true`) and so will return `false` and `expected` == `false.
* was `true` it will be swapped and so will return `true` and `expected == true`
So yes, you are right. I was pedantically substituting such that the `if` is checking against the old value of `mRedisplay` just like it always did. But yes, in this case `expected` is always equal to the result of `compare_exchange_strong`. I'd go either way here. Your version is more concise but is saying "Do a compare_exchange_strong and if that changed it...." versus my "Do a compare_exchange_strong and if `mRedisplay` was `true`...". I don't know which is clearer.

Reply:
The main advantage of my proposal is it more closely matches the flow of the old code.  But you are the one submitting, so I'd prefer it be whatever you are more confident in :>

Reply:
I like your conciseness. I hadn't noticed what you noticed; I was just mechanically porting it.

I'll change it to 
```
bool expected = true;
if (mRedisplay.compare_exchange_strong(expected, false)) {
    // mRedisplay was true and was just changed to false
```
Or should I put this into the if? I forget what C++ version we target... That would look like
```
if (bool expected = true;
    mRedisplay.compare_exchange_strong(expected, false)) {
    // mRedisplay was true and was just changed to false
```

Reply:
That would require C++17... I'm actually not sure what version we are at either, but C++17 is pretty new, so I'd avoid it unless we had a serious need (like constexpr if in templates!) rather than a slightly cleaner if :>